src/Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/Outflow.hpp

```diff
@@ -0,0 +1,135 @@
+namespace grmhd::ValenciaDivClean::BoundaryConditions {
+/// A `BoundaryCondition` that only verifies that all characteristic speeds are
+/// directed out of the domain; no boundary data is altered by this boundary
+/// condition.
+class Outflow final : public BoundaryCondition {
+ public:
+  using options = tmpl::list<>;
+  static constexpr Options::String help{
+      "Outflow boundary condition that only verifies the characteristic speeds "
+      "are all directed out of the domain."};
+
+  Outflow() = default;
+  Outflow(Outflow&&) noexcept = default;
+  Outflow& operator=(Outflow&&) noexcept = default;
+  Outflow(const Outflow&) = default;
+  Outflow& operator=(const Outflow&) = default;
+  ~Outflow() override = default;
+
+  explicit Outflow(CkMigrateMessage* msg) noexcept;
+
+  WRAPPED_PUPable_decl_base_template(
+      domain::BoundaryConditions::BoundaryCondition, Outflow);
+
+  auto get_clone() const noexcept -> std::unique_ptr<
+      domain::BoundaryConditions::BoundaryCondition> override;
+
+  static constexpr evolution::BoundaryConditions::Type bc_type =
+      evolution::BoundaryConditions::Type::Outflow;
+
+  void pup(PUP::er& p) override;
+
+  using dg_interior_evolved_variables_tags = tmpl::list<>;
+  using dg_interior_temporary_tags = tmpl::list<
+      gr::Tags::Shift<3, Frame::Inertial, DataVector>,
+      gr::Tags::Lapse<DataVector>,
+      gr::Tags::InverseSpatialMetric<3, Frame::Inertial, DataVector>>;
+  using dg_gridless_tags = tmpl::list<hydro::Tags::EquationOfStateBase>;
+  using dg_interior_primitive_variables_tags =
+      tmpl::list<hydro::Tags::RestMassDensity<DataVector>,
+                 hydro::Tags::SpecificInternalEnergy<DataVector>,
+                 hydro::Tags::SpecificEnthalpy<DataVector>,
+                 hydro::Tags::SpatialVelocity<DataVector, 3>,
+                 hydro::Tags::LorentzFactor<DataVector>,
+                 hydro::Tags::MagneticField<DataVector, 3>>;
+
+  template <typename EquationOfState>
+  static std::optional<std::string> dg_outflow(
+      const std::optional<tnsr::I<DataVector, 3, Frame::Inertial>>&
+          face_mesh_velocity,
+      const tnsr::i<DataVector, 3, Frame::Inertial>&
+          outward_directed_normal_covector,
+      const tnsr::I<DataVector, 3, Frame::Inertial>&
+      /*outward_directed_normal_vector*/,
+
+      const Scalar<DataVector>& rest_mass_density,
+      const Scalar<DataVector>& specific_internal_energy,
+      const Scalar<DataVector>& specific_enthalpy,
+      const tnsr::I<DataVector, 3, Frame::Inertial>& spatial_velocity,
+      const Scalar<DataVector>& lorentz_factor,
+      const tnsr::I<DataVector, 3, Frame::Inertial>& magnetic_field,
+
+      const tnsr::I<DataVector, 3, Frame::Inertial>& shift,
+      const Scalar<DataVector>& lapse,
+      const tnsr::II<DataVector, 3, Frame::Inertial>& inverse_spatial_metric,
+
+      const EquationOfState& equation_of_state) noexcept {
+    double min_speed = std::numeric_limits<double>::signaling_NaN();
```

Comment:
aren't the char speeds bounded by `\pm \alpha -\beta^i n_i` and so you only need to check those?

Reply:
Ah, yep, seems correct... That'll simplify matters considerably.